PR: https://github.com/odoo/odoo/pull/225653

From: 60b873ce7d68132be1109444157e11be07235582
From: soza-odoo
Date: 2025-10-27 10:03:08

Breaking data model changes scores: del:2 + add:0, change matches:
-    company_registry_placeholder = fields.Char(compute='_compute_company_registry_placeholder')
-    partner_company_registry_placeholder = fields.Char(compute='_compute_partner_company_registry_placeholder')

Total Changes: 31

[IMP] account:  remove duplicate company_registry_placeholder logic

Before:
    - We have two logics for the company placeholder, one in base and another
      one in account
    - We unify these two logics in this commit
    - We have one field 'partner_company_registry_placeholder' in res.partner
      In the account module, which stores the same value as
     'company_registry_placeholder'

After:
    - We remove that field in the account and rely on the related field from the
      base.
    - The placeholder field is now computed using a unified logic across all
      localizations.
    - We removed that 'partner_company_registry_placeholder'  field.

 The idea is that the compute should work by itself for any possible countries
 that have an entry in '_ref_company_registry' map, without the need
 of any extra steps.

task-5064900

closes odoo/odoo#225653

Related: odoo/upgrade#8412
Signed-off-by: Soham Zadafiya (soza) <soza@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account/models/company.py
+++ b/addons/account/models/company.py
@@ -11,7 +11,6 @@ from odoo.tools.mail import is_html_empty
 from odoo.tools.misc import format_date
 from odoo.addons.account.models.account_move import MAX_HASH_VERSION
 from odoo.addons.account.models.product import ACCOUNT_DOMAIN
-from odoo.addons.account.models.partner import _ref_company_registry
 from odoo.addons.base_vat.models.res_partner import _ref_vat
 from odoo.fields import Domain
 
@@ -286,7 +285,6 @@ class ResCompany(models.Model):
         help="Default on whether the sales price used on the product and invoices with this Company includes its taxes."
     )
     company_vat_placeholder = fields.Char(compute='_compute_company_vat_placeholder')
-    company_registry_placeholder = fields.Char(compute='_compute_company_registry_placeholder')
 
     income_account_id = fields.Many2one(
         comodel_name='account.account',
@@ -1123,15 +1121,6 @@ class ResCompany(models.Model):
 
             company.company_vat_placeholder = placeholder
 
-    @api.depends('country_id', 'account_fiscal_country_id')
-    def _compute_company_registry_placeholder(self):
-        """ Provides a dynamic placeholder on the company registry field for countries that may need it.
-        Add your country and the value you want in the _ref_company_registry map in the partner.py file.
-        """
-        for company in self:
-            country_code = (company.account_fiscal_country_id or company.country_id).code or ''
-            company.company_registry_placeholder = _ref_company_registry.get(country_code.lower(), '')
-
     def _set_category_defaults(self):
         for company in self:
             self.env['ir.default'].set('product.category', 'property_account_expense_categ_id', company.expense_account_id.id, company_id=company.id)

--- a/addons/account/models/partner.py
+++ b/addons/account/models/partner.py
@@ -19,6 +19,21 @@ _logger = logging.getLogger(__name__)
 _ref_company_registry = {
     'jp': '7000012050002',
     'dk': '58403288',
+
+    # Using the same placeholder value for France and all its overseas territories/collectivities
+    'fr': '33417522101010',
+    'pf': '33417522101010',
+    'mf': '33417522101010',
+    'mq': '33417522101010',
+    'nc': '33417522101010',
+    're': '33417522101010',
+    'gf': '33417522101010',
+    'gp': '33417522101010',
+    'tf': '33417522101010',
+    'bl': '33417522101010',
+    'pm': '33417522101010',
+    'yt': '33417522101010',
+    'wf': '33417522101010',
 }
 
 
@@ -320,7 +335,6 @@ class ResPartner(models.Model):
     fiscal_country_codes = fields.Char(compute='_compute_fiscal_country_codes')
     fiscal_country_group_codes = fields.Json(compute='_compute_fiscal_country_group_codes')
     partner_vat_placeholder = fields.Char(compute='_compute_partner_vat_placeholder')
-    partner_company_registry_placeholder = fields.Char(compute='_compute_partner_company_registry_placeholder')
     duplicate_bank_partner_ids = fields.Many2many(related="bank_ids.duplicate_bank_partner_ids")
 
     @api.depends('company_id')
@@ -1007,13 +1021,13 @@ class ResPartner(models.Model):
             partner.partner_vat_placeholder = placeholder
 
     @api.depends('country_id')
-    def _compute_partner_company_registry_placeholder(self):
+    def _compute_company_registry_placeholder(self):
         """ Provides a dynamic placeholder on the company registry field for countries that may need it.
         Add your country and the value you want in the _ref_company_registry map.
         """
         for partner in self:
             country_code = partner.country_id.code or ''
-            partner.partner_company_registry_placeholder = _ref_company_registry.get(country_code.lower(), '')
+            partner.company_registry_placeholder = _ref_company_registry.get(country_code.lower(), '')
 
     def _compute_account_move_count(self):
         # retrieve all children partners and prefetch 'parent_id' on them
